newIDE/app/src/VariablesList/VariablesEditorDialog.js

```diff
@@ -3,35 +3,58 @@ import FlatButton from 'material-ui/FlatButton';
 import Dialog from '../UI/Dialog';
 import { withSerializableObject } from '../Utils/SerializableObjectEditorContainer';
 import VariablesList from './index';
+
 const gd = global.gd;
 
 export class VariablesEditorDialog extends Component {
   render() {
+    const {
+      onCancel,
+      onApply,
+      open,
+      onEditObjectVariables,
+      titleMessage,
+      emptyExplanationMessage,
+      emptyExplanationSecondMessage,
+      variablesContainer,
+    } = this.props;
     const actions = [
-      <FlatButton label="Cancel" onClick={this.props.onCancel} />,
+      <FlatButton label="Cancel" onClick={onCancel} key={'Cancel'} />,
       <FlatButton
         label="Apply"
         primary
         keyboardFocused
-        onClick={this.props.onApply}
+        onClick={onApply}
+        key={'Apply'}
       />,
     ];
+    const secondaryActions =
+      onEditObjectVariables === undefined ? (
+        ''
```

Comment:
Use `null` instead of an empty string. `null` will render nothing in React.
`const secondaryActions = onEditObjectVariables  ? (FlatButton) : null;`